### DISCUSSION TITLE
is it to possible to get & change certificate pins using any mitmproxy API

### DISCUSSION THREAD

**Discussion by long106** [mitmproxy/mitmproxy#7065] [Intercept Everything!]
Hello all,

Is there a way to get & change the exact sha256 pin from any mitmproxy API?
The goal here is to bypass the certificate pinning of tiktok and see actual https traffic
ex: 
![image](https://github.com/user-attachments/assets/ae98d415-01ac-4d38-9eb2-46918aad9eff)

The test devices is an iphone--so the approaches using iptable does not work (https://github.com/mitmproxy/mitmproxy/discussions/6593)

**Comment by Prinzhorn** [comment]
If it was this trivial then it would be a worthless protection. The client will calculate the hash and compare it with the stored/pinned one. If you want to bypass pinning you need to patch the app to remove that logic. See https://docs.mitmproxy.org/stable/concepts-certificates/#certificate-pinning for a list of tools.
